controllers/controllers/workloads/shared.go

```diff
@@ -38,29 +44,173 @@ func setStatusConditionOnLocalCopy(conditions *[]metav1.Condition, conditionType
 	})
 }
 
-func createSubnamespaceAnchor(ctx context.Context, client client.Client, req ctrl.Request, object client.Object, labels map[string]string) (v1alpha2.SubnamespaceAnchor, error) {
-	anchor := v1alpha2.SubnamespaceAnchor{
+func createOrPatchNamespace(ctx context.Context, client client.Client, log logr.Logger, object client.Object, labels map[string]string) error {
+	namespace := &corev1.Namespace{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      req.Name,
-			Namespace: req.Namespace,
-			Labels:    labels,
-			OwnerReferences: []metav1.OwnerReference{
-				{
-					APIVersion: APIVersion,
-					Kind:       object.GetObjectKind().GroupVersionKind().Kind,
-					Name:       object.GetName(),
-					UID:        object.GetUID(),
-				},
-			},
+			Name: object.GetName(),
+		},
+	}
+
+	result, err := controllerutil.CreateOrPatch(ctx, client, namespace, func() error {
+		if namespace.ObjectMeta.Labels == nil {
+			namespace.ObjectMeta.Labels = make(map[string]string)
+		}
+
+		for key, value := range labels {
+			namespace.ObjectMeta.Labels[key] = value
+		}
+		return nil
+	})
+
+	if err != nil {
+		return err
+	}
+
+	log.Info(fmt.Sprintf("Namespace/%s %s", object.GetName(), result))
+	return nil
+}
+
+func propagateSecrets(ctx context.Context, client client.Client, log logr.Logger, object client.Object, secretName string) error {
```

Comment:
Here the `object` argument is the source cforg/cfspace, in order to make that explicit I would rename the `object` argument to something like `sourceOrgSpace`